lib/tasks/bundle_eval.rake

```diff
@@ -0,0 +1,194 @@
+        first_bundle.patients.each do |first_patient|
+          second_patient = second_bundle.patients.where(givenNames: first_patient.givenNames, familyName: first_patient.familyName).first
+          # Skip if the same patient can't be found in both bundles
+          next unless second_patient
+
+          first_patient.calculation_results.each do |fp_cr|
+            measure_id = second_bundle.measures.where(cms_id: fp_cr.measure.cms_id).first.id
+            sp_cr = second_patient.calculation_results.where(measure_id: measure_id, population_set_key: fp_cr.population_set_key).first
+            calc_diffs = sp_cr.nil? ? true : false
+            calc_diffs ||= result_different?(fp_cr, sp_cr)
+            csv << [fp_cr.measure.cms_id, first_patient.givenNames[0], first_patient.familyName] if calc_diffs
+          end
+        end
+        second_bundle.patients.each do |second_patient|
+          first_patient = first_bundle.patients.where(givenNames: second_patient.givenNames, familyName: second_patient.familyName).first
+          # Skip if the same patient can't be found in both bundles
+          next unless first_patient
+
+          second_patient.calculation_results.each do |sp_cr|
+            measure_id = first_bundle.measures.where(cms_id: sp_cr.measure.cms_id).first.id
+            fp_cr = first_patient.calculation_results.where(measure_id: measure_id, population_set_key: sp_cr.population_set_key).first
+            calc_diffs = sp_cr.nil? ? true : false
+            calc_diffs ||= result_different?(sp_cr, fp_cr)
+            csv << [sp_cr.measure.cms_id, second_patient.givenNames[0], second_patient.familyName] if calc_diffs
+          end
+        end
+      end
+    end
+
+    desc %(Evaluate a bundle for potential disconnects between Diagnosis and Encounter Diagnosis.
+
+    You must identify the bundle version:
+
+    $ rake bundle:eval:code_crosswalk[xxxx.x.x])
+    task :find_diagnosis_differences, [:bundle_version] => :setup do |_, args|
+      all_vs = {}
+      CSV.open('tmp/find_diagnosis_differences.csv', 'w') do |csv|
+        bundle = Bundle.where(version: args.bundle_version).first
+        bundle.patients.each do |bp|
+          # Array of all valuesets used in a patient's condition list
+          bp_condition_vs_list = []
+          # Array of all valuesets used in a patient's encounters principalDiagnosis and diagnoses
+          en_encounter_dx_vs_list = []
+          bp.qdmPatient.conditions.each do |c|
+            c['dataElementCodes'].each do |dec|
+              # check if snomed
+              next unless dec['codeSystemOid'] == '2.16.840.1.113883.6.96'
+
+              ValueSet.where('concepts.code' => dec['code']).each do |vs|
+                bp_condition_vs_list << vs.oid
+              end
+            end
+          end
+          bp.qdmPatient.get_data_elements('encounter', 'performed').each do |c|
```

Comment:
change c to e or something to indicate that we're iterating over an encounter list